diff -up lxml-3.2.1/src/lxml/iterparse.pxi.orig lxml-3.2.1/src/lxml/iterparse.pxi
--- lxml-3.2.1/src/lxml/iterparse.pxi.orig	2013-03-29 21:50:04.000000000 +0100
+++ lxml-3.2.1/src/lxml/iterparse.pxi	2013-07-11 17:33:02.859572207 +0200
@@ -457,7 +457,10 @@ cdef class iterparse(_BaseParser):
         return context
 
     cdef _close_source(self):
-        if self._source is None or not self._close_source_after_read:
+        if self._source is None:
+            return
+        if not self._close_source_after_read:
+            self._source = None 
             return
         try:
             close = self._source.close
